vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -73,16 +73,13 @@
       keys: "{{ configmap.resources[0].data.keys() | map('regex_replace', 'ENDPOINT', 'URL') | list }}"
       values: "{{ configmap.resources[0].data.values() | list }}"
 
-
-  - name: Backup ironic credentials for re-use when pods are re-created during the upgrade process
-    kubernetes.core.k8s_info:
-      api_version: v1
-      kind: Secret
-      namespace: "{{ NAMEPREFIX }}-system"
-      label_selectors:
-        - cluster.x-k8s.io/provider = infrastructure-metal3
-      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    register: ironic_credentials
+  - name: Backup secrets for re-use when pods are re-created during the upgrade process
+    shell: |
+           kubectl get secrets -n {{ NAMEPREFIX }}-system -o json |
+           jq '.items[]|del(.metadata|.managedFields,.uid,.resourceVersion)' > /tmp/secrets.with.values.yaml
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    ignore_errors: yes
```

Comment:
why do we ignore errors and what's the problem with `kubernetes.core.k8s_info` ?

Reply:
shell format is borrowed from previous version of the upgrade file, because ansible version of doing the task has problem with fetching ironic credentials using label_selectors and registered **ironic_credentials** var becomes empty. I simply reverted back to shell in short

Reply:
Okay, I'm still not clear why we are ignoring errors here and I hope it won't cause any issues where we don't stop the job when there is an error.

Regarding label_selectors, you could also check filter plugin https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/filter_plugins/status.py.

But to unblock broken upgrade tests I am going to approve it now.